src/strands/agent/agent.py

```diff
@@ -381,21 +382,22 @@ def __call__(self, prompt: str, **kwargs: Any) -> AgentResult:
                 - metrics: Performance metrics from the event loop
                 - state: The final state of the event loop
         """
-        callback_handler = kwargs.get("callback_handler", self.callback_handler)
 
-        self._start_agent_trace_span(prompt)
+        async def acall() -> AgentResult:
+            callback_handler = kwargs.get("callback_handler", self.callback_handler)
 
-        try:
             events = self._run_loop(callback_handler, prompt, kwargs)
-            for event in events:
+            async for event in events:
```

Comment:
Is it possible to call `stream_async` here or unify it so that they're both just plumbing to the same place?

And with #289 we'll now removing the callback_handler override, so I think this gets simpler overall.  Maybe coordinate with @Unshure to see what else simplifies down